sdk/keyvault/keyvault-secrets/src/index.ts

```diff
@@ -68,93 +68,142 @@ export { ProxyOptions, RetryOptions, TelemetryOptions };
-   */
-  protected readonly credential: ServiceClientCredentials | TokenCredential;
-  private readonly client: KeyVaultClient;
+  private async *listDeletedSecretsPage(
+    continuationState: PageSettings,
+    options?: ListSecretsOptions
+  ): AsyncIterableIterator<SecretAttributes[]> {
+    if (continuationState.continuationToken == null) {
+      const optionsComplete: KeyVaultClientGetSecretsOptionalParams = {
+        maxresults: continuationState.maxPageSize,
+        ...(options && options.requestOptions ? options.requestOptions : {})
+      };
+      const currentSetResponse = await this.client.getDeletedSecrets(
+        this.vaultBaseUrl,
+        optionsComplete
+      );
+      continuationState.continuationToken = currentSetResponse.nextLink;
+      yield currentSetResponse.map(this.getSecretFromSecretBundle);
+    }
+    while (continuationState.continuationToken) {
+      const currentSetResponse = await this.client.getDeletedSecretsNext(
+        continuationState.continuationToken,
+        options
+      );
+      continuationState.continuationToken = currentSetResponse.nextLink;
+      yield currentSetResponse.map(this.getSecretFromSecretBundle);
+    }
+  }
 
-  /**
-   * Creates an instance of SecretsClient.
-   * @param {string} url the base url to the key vault.
-   * @param {ServiceClientCredentials | TokenCredential} The credential to use for API requests.
-   * @param {(Pipeline | NewPipelineOptions)} [pipelineOrOptions={}] Optional. A Pipeline, or options to create a default Pipeline instance.
-   *                                                                 Omitting this parameter to create the default Pipeline instance.
-   * @memberof SecretsClient
-   */
-  constructor(
-    url: string,
-    credential: ServiceClientCredentials | TokenCredential,
-    pipelineOrOptions: Pipeline | NewPipelineOptions = {}
-  ) {
-    this.vaultBaseUrl = url;
-    this.credential = credential;
-    if (isNewPipelineOptions(pipelineOrOptions)) {
-      this.pipeline = SecretsClient.getDefaultPipeline(credential, pipelineOrOptions);
+  private async *listDeletedSecretsAll(
+    options?: ListSecretsOptions
+  ): AsyncIterableIterator<SecretAttributes> {
+    const f = {};
+
+    for await (const page of this.listDeletedSecretsPage(f, options)) {
+      for (const item of page) {
+        yield item;
+      }
+    }
+  }
+
+  private getSecretFromSecretBundle(secretBundle: SecretBundle): Secret {
+    const parsedId = parseKeyvaultEntityIdentifier("secrets", secretBundle.id);
+
+    let resultObject;
+    if (secretBundle.attributes) {
+      resultObject = {
+        ...secretBundle,
+        ...parsedId,
+        ...secretBundle.attributes
+      };
+      delete resultObject.attributes;
     } else {
-      this.pipeline = pipelineOrOptions;
+      resultObject = {
+        ...secretBundle,
+        ...parsedId
+      };
     }
 
-    this.client = new KeyVaultClient(credential, "7.0", this.pipeline);
```

Comment:
Why is this line changing?

Reply:
It's not! It just moved, the diff tool isn't that smart.